gluonnlp/data/sampler.py

```diff
@@ -194,9 +207,16 @@ def __init__(self, lengths, batch_size, num_buckets=10, bucket_keys=None,
                                    if len(sample_ids) > 0]
         scale_up_keys = [key if self._single_element else sum(key) for key in self._bucket_keys]
         max_scale_up_key = max(scale_up_keys)
-        self._bucket_batch_sizes = [max(int(max_scale_up_key / float(scale_up_key)
-                                            * self._ratio * batch_size), batch_size)
-                                    for scale_up_key in scale_up_keys]
+        if batch_type == 'sentence':
+            self._bucket_batch_sizes = [max(int(max_scale_up_key / float(scale_up_key)
+                                                * self._ratio * batch_size), batch_size)
+                                        for scale_up_key in scale_up_keys]
+        elif batch_type == 'word':
+            bucket_average_lengths = _bucket_average_lengths(self._bucket_sample_ids, self._lengths)
+            self._bucket_batch_sizes = [max(int(batch_size / float(average_length)), 1)
+                                        for average_length in bucket_average_lengths]
```

Comment:
the scale_up_keys are not longer used if this mode is chosen. the API doc should explain the ratio switch doesn't take effect if use_bucket_avg is true, and give warning when both are specified.